docs/src/api-private.md

```diff
@@ -18,7 +18,7 @@ TimeZones.TZData.tzdata_version_dir
 TimeZones.TZData.tzdata_version_archive
 TimeZones.TZData.read_news
 TimeZones.TZData.compile!
-TimeZones.TZData.tryparse_dayofmonth
+TimeZones.TZData.tryparse_dayofmonth_function
```

Comment:
Seems like a weird change to just add `function` to the end

Reply:
To make this more clear I would prefer `gen_tryparse_dayofmonth_function` but it is already a really long function name so no need to change.

Reply:
While revising this code I thought of `tryparse_dayofmonth` as a function that would return an integer. I added "_function" to be more explicit that a function is returned. Additionally, I think adding a "gen" prefix doesn't add much since `parse` here means: convert a string into a Julia representation. Also, the internals here could be changed to return a pre-created function instead of a closure.